flake.nix

```diff
@@ -29,28 +20,37 @@
       inputs.nixpkgs.follows = "nixpkgs";
       inputs.flake-utils.follows = "devshell/flake-utils";
     };
-    nix-filter.url = "github:numtide/nix-filter";
     flake-compat.url = "github:nix-community/flake-compat";
+    nix-filter.url = "github:numtide/nix-filter";
     nix-github-actions = {
       url = "github:nix-community/nix-github-actions";
       inputs.nixpkgs.follows = "nixpkgs";
     };
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    treefmt-nix = {
+      url = "github:numtide/treefmt-nix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
   };
 
-  outputs = inputs @ {flake-parts, ...}:
-    flake-parts.lib.mkFlake
-    {
+  outputs = inputs: let
+    inherit (inputs.nixpkgs) lib;
+  in
+    inputs.blueprint {
       inherit inputs;
+      prefix = "nix/";
+      nixpkgs.config = {
+        allowUnfreePredicate = pkg: builtins.elem (lib.getName pkg) ["terraform"];
+      };
     }
-    {
-      imports = [
-        ./nix
-      ];
-      systems = [
-        "x86_64-linux"
-        "aarch64-linux"
-        "x86_64-darwin"
-        "aarch64-darwin"
-      ];
+    // {
+      githubActions = inputs.nix-github-actions.lib.mkGithubMatrix {
+        checks =
+          lib.getAttrs [
+            "x86_64-linux"
+            "x86_64-darwin"
+          ]
+          inputs.self.checks;
+      };
```

Comment:
Eventually this could also be handled by blueprint. Adding a reference back to here:
https://github.com/numtide/blueprint/issues/21